Userland/Libraries/LibVideo/PlaybackManager.cpp

```diff
@@ -186,112 +208,140 @@ void PlaybackManager::seek_to_timestamp(Time target_timestamp, SeekMode seek_mod
 #endif
 
-    auto enqueue_error = [&](DecoderError&& error, Time timestamp) {
-        dbgln_if(PLAYBACK_MANAGER_DEBUG, "Enqueued decoder error: {}", error.string_literal());
-        m_frame_queue->enqueue(FrameQueueItem::error_marker(move(error), timestamp));
-    };
-
-#define TRY_OR_ENQUEUE_ERROR(expression, timestamp)                                                  \
-    ({                                                                                               \
-        auto&& _temporary_result = (expression);                                                     \
-        if (_temporary_result.is_error()) {                                                          \
-            enqueue_error(_temporary_result.release_error(), (timestamp));                           \
-            return false;                                                                            \
-        }                                                                                            \
-        static_assert(!::AK::Detail::IsLvalueReference<decltype(_temporary_result.release_value())>, \
-            "Do not return a reference from a fallible expression");                                 \
-        _temporary_result.release_value();                                                           \
-    })
+    FrameQueueItem item_to_enqueue;
+
+    while (item_to_enqueue.is_empty()) {
+        // Get a sample to decode.
+        auto sample_result = ({
+            Threading::MutexLocker demuxer_locker(m_demuxer_mutex);
+            m_demuxer->get_next_video_sample_for_track(m_selected_video_track);
+        });
+        if (sample_result.is_error()) {
+            item_to_enqueue = FrameQueueItem::error_marker(sample_result.release_error(), Time::min());
+            break;
+        }
+        auto sample = sample_result.release_value();
 
-    auto frame_sample = TRY_OR_ENQUEUE_ERROR(m_demuxer->get_next_video_sample_for_track(m_selected_video_track), Time::min());
-    OwnPtr<VideoFrame> decoded_frame = nullptr;
-    while (!decoded_frame) {
-        TRY_OR_ENQUEUE_ERROR(m_decoder->receive_sample(frame_sample->data()), frame_sample->timestamp());
+        // Submit the sample to the decoder.
+        auto decode_result = m_decoder->receive_sample(sample->data());
+        if (decode_result.is_error()) {
+            item_to_enqueue = FrameQueueItem::error_marker(decode_result.release_error(), sample->timestamp());
+            break;
+        }
 
+        // Retrieve the last available frame to present.
+        OwnPtr<VideoFrame> decoded_frame = nullptr;
         while (true) {
             auto frame_result = m_decoder->get_decoded_frame();
 
             if (frame_result.is_error()) {
-                if (frame_result.error().category() == DecoderErrorCategory::NeedsMoreInput)
+                if (frame_result.error().category() == DecoderErrorCategory::NeedsMoreInput) {
                     break;
+                }
```

Comment:
Braces should be left out for single-line if-blocks such as these.

Reply:
Went through and removed those and some others I added in that commit.